examples/tasks/async_compute.rs

```diff
@@ -0,0 +1,121 @@
+                    let duration = Duration::from_secs_f32(rng.gen_range(0.05..0.2));
+                    while Instant::now() - start_time < duration {
+                        // Simulating doing hard compute work generating translation coords!
+                    }
+
+                    // Such hard work, all done!
+                    println!("Done generating translation coords x: {} y: {} z: {}", x, y, z);
+
+                    Transform::from_translation(Vec3::new(x as f32, y as f32, z as f32))
+                });
+
+                // Spawn new entity, tag it with Marker as a component,
+                // and add our new task as a component
+                commands.spawn()
+                    .insert(Marker)
+                    .insert(task);
+            }
+        }
+    }
+}
+
+/// This system queries for entities that have both our Marker component
+/// as well as a Task<Transform> component. It polls the tasks to see if they're
+/// complete. If the task is complete it takes the result, adds a new PbrBundle of components to the
+/// entity using the result from the task's work, and removes the task component from the entity.
+fn handle_tasks(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    mut our_entity_tasks: Query<(Entity, &mut Task<Transform>), With<Marker>>
+) {
+    for (entity, mut task) in our_entity_tasks.iter_mut() {
+
+        if let Some(transform) = future::block_on(future::poll_once(&mut *task)) {
+
+            // Normally we would add our mesh and material assets once
+            // and store the handle, but that's for another example
+            let box_mesh = meshes.add(Mesh::from(shape::Cube { size: 0.25 }));
+            let box_material = materials.add(Color::rgb(1.0, 0.2, 0.3).into());
+
+            // Add our new PbrBundle of components to our tagged entity
+            commands.entity(entity).insert_bundle(PbrBundle {
+                mesh: box_mesh.clone(),
+                material: box_material.clone(),
+                transform,
+                ..Default::default()
+            });
+
+            // Task is complete, so remove task component from entity
+            commands.entity(entity).remove::<Task<Transform>>();
+        }
+    }
+}
+
+fn main() {
+    App::build()
+        .insert_resource(Msaa { samples: 4 })
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup_env.system())
+        .add_startup_system(spawn_tasks.system())
+        .add_system(handle_tasks.system())
+        .run();
+}
+
+/// This system IS NOT part of the example, it's only
+/// used to setup the light and camera for the environment
+fn setup_env(mut commands: Commands) {
+
+    // Used to center camera on spawned cubes
+    let offset = || {
+        if NUM_CUBES % 2 == 0 {
+            (NUM_CUBES / 2) as f32 - 0.5
+        } else {
+            (NUM_CUBES / 2) as f32
+        }

```

Comment:
`NUM_CUBES` is a constant so shouldn't need an `if`/`else` chain 🙂 

```rust
let offset = (NUM_CUBES / 2) as f32 - 0.5;
```

Reply:
If someone using the example changes to the number of cubes to an odd number, then the camera would be off by ``-0.5``

Reply:
Hmmm, that's fair. It's just a little weird with it being a `const`, however since it's an example and someone could change it, that may be ok.

Reply:
> Hmmm, that's fair. It's just a little weird with it being a `const`, however since it's an example and someone could change it, that may be ok.

I wanted offset to be ``const fn`` but no floats in ``const fn``s. I figured it would be okay putting it in the system commented as not part of the example topic. But I have no problem removing it if others feel otherwise.

Reply:
Well it actually doesn't have to be a function, you can just use a scope.
```rust
 let offset = {
        if NUM_CUBES % 2 == 0 {
            (NUM_CUBES / 2) as f32 - 0.5
        } else {
            (NUM_CUBES / 2) as f32
        };
```

Reply:
Changed in 6dcd5d1